### ISSUE TITLE
Black boxes

### ISSUE THREAD

**Issue by manio** [any1/wlvncc#16]
Hello,
I am trying to reach a reasonable VNC client performance using RaspberryPi. I am running tigervnc server on a remote server. Your project is the first one I've tested which works with decent level of scrolling speed eg in remote web browser.
I've tested accelerated xorg with vncviewer and also vncviewer running under xwayland but the speed is poor.
But the main problem is that when I am using your client, then I can see some black boxes which are "flying" around the screen (in random locations). I am using enlightenment on remote session. Do you have any tips for me to get rid of it?
Is it a known problem? Maybe I can adjust something on the server side in VNC protocol?

Besides rpi I was also testing it on regular PC (x86_64) and the same problem appear (I was starting it from sway and IIRC weston).

**Comment by any1** [comment]
You could try using software rendering. That's the `-s` flag. Also, try going through some different encoding methods. See `wlvncc --help` for a list of selectable encodings.

**Comment by manio** [comment]
Yay! software rendering helps! No more black boxes. And it seems it is working quite nice (maybe not as fast as with hardware rendering but it is still usable).
I was also testing all encodings (without -s flag) and the black boxes was still there.

I think I could prepare you some test environment (a clean enlightenment container + tigerVNC on this host) and provide you connection details (via private mail). You could then connect directly to this VNC server and maybe look at this problem. Are you interested in this?

**Comment by any1** [comment]
I just tried doing some scrolling in a browser with wayvnc and was unable to reproduce the issue. If you could provide me with a Dockerfile that I can test at my own leisure, that would be great.

**Comment by manio** [comment]
Andri, I've sent you an email but I can see you're using google server. They don't like my e-mail domain - please check your Spam folder :)

**Comment by any1** [comment]
Yeah, sorry about google email. It's a battle I've yet to take on. Maybe I'll move to fastmail or something in the future. I see nothing from you in my spam directory. There's also IRC. Check out #wayvnc on libera net.

**Comment by manio** [comment]
It seems that the problem is related to the hardware rendering and may be related to specific driver. The issue cannot be reproduced on @any1 configuration.
Thank you very much for helping (especially pointing me a software rendering switch).
Closing.

**Comment by manio** [comment]
Only for future reference: https://github.com/any1/wlvncc/tree/fix-damage-tracking

**Comment by any1** [comment]
I thought we established that this change doesn't do anything?

**Comment by manio** [comment]
Correct. I only put this here for remembering that the fix attempt was at this specific tree.

**Comment by any1** [comment]
Rendering has seen a lot of work, so this is probably not relevant any more. Closing for now.
